### PR TITLE
feat: add aws terraform example

### PR DESCRIPTION
**Pull request by TimJones**
Adds the Terraform example for a Talos cluster on AWS.

Signed-off-by: Tim Jones <tim.jones@siderolabs.com>

### CODE REVIEW

**Review by smira — APPROVED** [review]

File: examples/terraform/aws/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+## Usage
+
+To create a default cluster, this should be as simple as `terraform apply`.
+This will create a cluster called `talos-aws-example` with 3 control plane nodes and a single worker in the default AWS region.
+By default, the instances will be `c5.large`, with 2 VPU and 4GB RAM each.
+If different specs or regions are required, override them through command line with the `-var` flag or by creating a varsfile and overriding with `-var-file`.
+Destroying the cluster should, again, be a simple `terraform destroy`.
+
+Getting the kubeconfig and talosconfig for this cluster can be done with `terraform output -raw kubeconfig > <desired-path-and-filename>` and `terraform output -raw talosconfig > <desired-path-and-filename>`
```
Comment (frezbo) [inline examples/terraform/aws/README.md:19]: do we need to add a note for the user saying that machineconfig could potentially be available to workloads if not secured properly?

**Review by frezbo — APPROVED** [review]

File: examples/terraform/aws/README.md (hunk 1 above)
Comment (rsmitty) [reply examples/terraform/aws/README.md:19]: I thought we were going to apply-config to add the machine config after launching the VMs as opposed to bundling it in at create time.

Comment (TimJones) [reply examples/terraform/aws/README.md:19]: Did we?? I missed/forgot that :man_facepalming:. Will fix.

**Review by rsmitty — APPROVED** [review]

**Pull request merged**